src/modules/navigator/mission.cpp

```diff
@@ -1019,7 +1019,6 @@ Mission::set_mission_items()
 
 					_mission_item.altitude = altitude;
 					_mission_item.altitude_is_relative = false;
-					_mission_item.nav_cmd = NAV_CMD_WAYPOINT;
```

Comment:
Couldn't this removal of the default cmd to type WAYPOINT cause problems? Why is it necessary in the context of this PR?

Reply:
@sfuhrer Because if you scroll a few lines up you will see that we are in the context of NAV_CMD_VTOL_LAND and we should not need to tamper with the nav command just to achieve an expected behavior. Now, the position controller understands position setpoint type land so we can let it handle it. That's much cleaner IMO.

Reply:
right, I was not at the right location, all good!